pkg/controllers/rollout/controller_integration_test.go

```diff
@@ -388,11 +388,11 @@ var _ = Describe("Test the rollout Controller", func() {
 		Consistently(func() bool {
 			for _, binding := range bindings {
 				err := k8sClient.Get(ctx, types.NamespacedName{Name: binding.GetName()}, binding)
-				if err != nil {
-					return false
-				}
-				if binding.Spec.State == fleetv1beta1.BindingStateBound {
+				if err == nil && binding.Spec.State == fleetv1beta1.BindingStateBound {
 					return false
+				} else if err != nil && !apierrors.IsNotFound(err) {
+					// do not return false on api error to reduce test flakyness
+					By(fmt.Sprintf("failed to get resource binding  %s", binding.Name))
```

Comment:
Would it better to ensure the bindings exist in an Eventaully block where we get the bindings before we run the Consistently block ?